### PR TITLE
Remove panicking ops from mempool code

### PR DESCRIPTION
**Pull request by akoptelov**

### CODE REVIEW

File: shell_automaton/src/mempool/mempool_effects.rs [hunk 1]
```diff
@@ -64,13 +66,12 @@ where
...
-                .unwrap()
-                .hash
-                .eq(&key.0)
+                .map(|h| h.hash.eq(&key.0))
+                .unwrap_or(false)
             {
                 let operation_hashes = value
                     .iter()
-                    .map(|op| op.message_typed_hash::<OperationHash>().unwrap())
+                    .filter_map(|op| op.message_typed_hash::<OperationHash>().map_or(None, Some))
```
Comment (binier) [inline shell_automaton/src/mempool/mempool_effects.rs:74]: Here you can use just `op.message_typed_hash::<OperationHash>().ok()`

File: shell_automaton/src/mempool/mempool_effects.rs [hunk 2]
```diff
@@ -101,14 +102,11 @@ where
...
-                            .state()
-                            .mempool
-                            .prevalidator
-                            .as_ref()
-                            .unwrap()
-                            .protocol
-                            .to_base58_check();
+                        let prevalidator = match store.state().mempool.prevalidator.as_ref() {
+                            Some(v) => v,
+                            None => continue,
```
Comment (binier) [inline shell_automaton/src/mempool/mempool_effects.rs:107]: probably we can just `return` here.

**Review by vlad9486 — APPROVED** [review]

File: shell_automaton/src/mempool/mempool_effects.rs [hunk 3]
```diff
@@ -357,12 +354,11 @@ where
             }
         }
         Action::MempoolGetOperations(MempoolGetOperationsAction { address }) => {
-            let peer = store
-                .state()
-                .mempool
-                .peer_state
-                .get(address)
-                .expect("enabling condition");
+            let peer = if let Some(peer) = store.state().mempool.peer_state.get(address) {
```
Comment (binier) [inline shell_automaton/src/mempool/mempool_effects.rs:357]: probably `match` a bit cleaner and more consistent with the codebase:

```
let peer = match store.state().mempool.peer_state.get(address) {
    Some(v) => v,
    None => return,
};
```

**Review by binier — APPROVED** [review]
Please check out those changes that I wrote in comments.

Approved, looks good.

**Pull request merged**